tests/unit/samplegen/test_integration.py

```diff
@@ -124,14 +126,206 @@ def test_generate_sample_service_not_found():
+    for p in [
+            "config_1.yaml",
+            "config_2.yaml",
+            "config_notes.txt",
+            "subdir/config_3.yaml",
+            "subdir/config_4.yaml",
+            "subdir/nested/config_5.yaml",
+    ]:
+        fs.create_file(path.join(directory, p), contents=good_contents)
+
+    for p in [
+            "bad_config_1.yaml",
+            "subdir/bad_config_2.yaml",
+            "subdir/nested/bad_config_3.yaml",
+    ]:
+        fs.create_file(path.join(directory, p), contents=bad_contents)
+
+    expected_paths = [
+        "sampleconfig/config_1.yaml",
+        "sampleconfig/config_2.yaml",
+        "sampleconfig/subdir/config_3.yaml",
+        "sampleconfig/subdir/config_4.yaml",
+        "sampleconfig/subdir/nested/config_5.yaml",
+    ]
+
+    actual_paths = sorted(gapic_utils.generate_all_sample_fpaths(directory))
+
+    assert actual_paths == expected_paths
+
+
+def test_generate_sample_config_fpaths_directories_no_configs(fs):
+    directory = 'sampleconfig'
+    for f in ['a.yaml', 'b.yaml']:
+        fs.create_file(path.join(directory, f))
+
+    actual_paths = list(gapic_utils.generate_all_sample_fpaths(directory))
+
+    assert not actual_paths
+
+
+def test_generate_sample_config_fpaths_not_yaml(fs):
+    expected_path = 'cfgs/sample_config.not_yaml'
+    fs.create_file(expected_path)
+
+    with pytest.raises(types.InvalidConfig):
+        list(gapic_utils.generate_all_sample_fpaths(expected_path))
+
+
+def test_generate_sample_config_fpaths_bad_contents(
+        fs,
+        # Note the typo: SampleConfigPronto
+        contents=dedent(
+            '''
+            ---
+            type: com.google.api.codegen.SampleConfigPronto
+            schema_version: 1.2.0
+            samples:
+            - service: google.cloud.language.v1.LanguageService
+            '''
+        )
+):
+    expected_path = 'cfgs/sample_config.yaml'
+    fs.create_file(expected_path, contents=contents)
+
+    with pytest.raises(types.InvalidConfig):
+        list(gapic_utils.generate_all_sample_fpaths(expected_path))
+
+
+def test_generate_sample_config_fpaths_bad_contents_old(fs):
+    test_generate_sample_config_fpaths_bad_contents(
+        fs,
+        contents=dedent(
+            '''
+            ---
+            type: com.google.api.codegen.SampleConfigProto
+            schema_version: 1.1.0
+            samples:
+            - service: google.cloud.language.v1.LanguageService
+            '''
+        )
+    )
+
+
+def test_generate_sample_config_fpaths_bad_contents_no_samples(fs):
+    test_generate_sample_config_fpaths_bad_contents(
+        fs,
+        contents=dedent(
+            '''
+            ---
+            type: com.google.api.codegen.SampleConfigProto
+            schema_version: 1.2.0
+            '''
+        )
+    )
+
+
+def test_generate_sample_config_partial_config(fs):
+    expected_path = 'sample.yaml'
+    fs.create_file(
+        expected_path,
+        # Note the typo: SampleConfigPront
```

Comment:
nit: "o" at the end

Do you want to keep the comment at the top of the typo'd YAML doc? Would make it clearer....

Reply:
I'm on the fence. If I leave it out of the yaml doc, then it gets subject to editor syntax highlighting and pops out more. If it's in the doc, it's closer to the source.